DcCore/DcCore/Helper/DcUtils.swift

```diff
@@ -83,6 +83,22 @@ public struct DcUtils {
         }
     }
 
+    public static func saveAnimatedImage(data: Data, suffix: String) -> String? {
+        guard let directory = try? FileManager.default.url(for: .documentDirectory, in: .userDomainMask,
+                                                           appropriateFor: nil, create: false) as NSURL else {
+                                                            return nil
+        }
+        do {
+            let timestamp = Double(Date().timeIntervalSince1970)
+            let path = directory.appendingPathComponent("\(timestamp).\(suffix)")
+            try data.write(to: path!)
```

Comment:
probably better to unwrap `path`.

In general it is possible to unwrap several optionals in a single `guard let` statement.
So you could youse the `guard let` in line 87 like this.

`guard let directory = try ? ..., let path = directory ... else { return }`